lib/doorkeeper/oauth/token_introspection.rb

```diff
@@ -167,13 +167,12 @@ def authorized_token_matches_introspected?
       end
 
       # If token doesn't belong to some client, then it is public.
-      # Otherwise in it required for token to be connected to the same client.
+      # Otherwise in it required for token to be connected to the same client by default,
+      # or by other configuration in Doorkeeper.configuration.allow_token_introspection.
       def authorized_for_client?
-        if @token.application
-          @token.application == authorized_client.application
-        else
-          true
-        end
+        return true unless @token.application
```

Comment:
I think it must be in configuration block default body because some developers may want to forbid introspection for **any** access token regardless it's publicity

Reply:
Yes, I will put original constraints of `authorized_for_client ?` method to block default body.